tests/js/client/restart/test-restart-replication2-view-recovery-cluster.js

```diff
@@ -0,0 +1,542 @@
+    const docs = [];
+    for (let i = 0; i < number; ++i) {
+      const doc = genDoc();
+      const {_key} = collection.save(doc);
+      docs.push({doc, _key});
+    }
+    for (let i = 0; i < updates; ++i) {
+      for (const {_key, doc} of docs) {
+        collection.replace(_key, doc);
+      }
+    }
+  };
+
+  const assertViewInSync = () => {
+    const result = db._query(`FOR doc IN ${viewName} SEARCH doc.c >= 0 OPTIONS {waitForSync: true} RETURN doc._key`).toArray();
+    const expectedResult = db._query(`FOR doc IN ${collection.name()} FILTER doc.c >= 0 RETURN doc._key`).toArray();
+
+    const uniqView = new Set(result);
+    const uniqCol = new Set(expectedResult);
+    if (result.length !== expectedResult.length) {
+      {
+        const dh = require("@arangodb/testutils/document-state-helper");
+        let intermediateCommitEntries = dh.getDocumentEntries(dh.mergeLogs(logs), "Insert");
+        require("console").error(JSON.stringify(intermediateCommitEntries));
+      }
+      require("console").error(`Counts missmatch expecting: ${expectedResult.length} got: ${result.length}`);
+      require("console").error(`Unique Counts expecting: ${uniqCol.size} got: ${uniqView.size}`);
+      if (uniqView.size < result.length) {
+        const tmp = new Set();
+        // Found duplicate keys
+        for (const i of result) {
+          if (tmp.has(i)) {
+            require("console").error(`Duplicate Key in result: ${i}`);
+          } else {
+            tmp.add(i);
+          }
+        }
+      }
+      for (const i of result) {
+        if (!uniqCol.has(i)) {
+          require("console").error(`Key in result not in collection: ${i}`);
+        }
+      }
+    }
+    assertEqual(result.length, expectedResult.length, `View: ${result.length}, Col: ${expectedResult.length}`);
+  };
+
+  return {
+    setUpAll,
+    tearDownAll,
+
+    setUp: setUpAnd(() => {
+      const setupInfo = setupCollection();
+      collection = setupInfo.collection;
+      shards = setupInfo.shards;
+      shardsToLogs = setupInfo.shardsToLogs;
+      logs = setupInfo.logs;
+
+      db._dropView(viewName);
+      db._createView(viewName, 'arangosearch', {});
+
+      var meta = { links: { [collection.name()]: { includeAllFields: true } } };
+      db._view(viewName).properties(meta);
+    }),
+
+    tearDown: tearDownAnd(() => {
+      dropCollection();
+    }),
+
+    /*
+     * Recover the view if the log is empty
+     */
+    [`testRecoverEmptyLog_${namePostfix}`]() {
+      insertDummyData(documentCount);
+      compactLogs();
+      lh.bumpTermOfLogsAndWaitForConfirmation(dbn, collection);
+      assertViewInSync();
+    },
+
+    /*
+    * Recover the view if the log is contains entries
+    */
+    [`testRecoverInsertsInLog_${namePostfix}`]() {
+      compactLogs();
+      insertDummyData(documentCount);
+      lh.bumpTermOfLogsAndWaitForConfirmation(dbn, collection);
+      assertViewInSync();
+    },
```

Comment:
Since `compactLogs()` doesn't really do anything (it's an empty for loop), I would argue that these two tests are the same 🤔